Kununu/ArchitectureSniffer/Configuration/Rules/MustExtend.php

```diff
@@ -4,48 +4,57 @@
 namespace Kununu\ArchitectureSniffer\Configuration\Rules;
 
 use InvalidArgumentException;
-use JsonException;
+use Kununu\ArchitectureSniffer\Configuration\ArchitectureLibrary;
+use Kununu\ArchitectureSniffer\Configuration\Group;
 use Kununu\ArchitectureSniffer\Configuration\Selector\InterfaceClassSelector;
 use Kununu\ArchitectureSniffer\Configuration\Selector\Selectable;
-use Kununu\ArchitectureSniffer\Configuration\Selectors;
+use PHPat\Test\Builder\Rule as PHPatRule;
 use PHPat\Test\PHPat;
 
-final readonly class MustExtend implements Rule
+final readonly class MustExtend extends AbstractRule
 {
-    public const string KEY = 'extends';
+    public static function createRule(
+        string $groupName,
+        ArchitectureLibrary $selectorsLibrary,
+    ): PHPatRule {
+        $includes = $selectorsLibrary->getIncludesByGroup($groupName);
+        $excludes = $selectorsLibrary->getExcludesByGroup($groupName);
+        $extensions = self::checkIfNotInterfaceSelectors(
+            $selectorsLibrary->getTargetByGroup($groupName, Group::EXTENDS_KEY)
+        );
+        $extensionExcludes = $selectorsLibrary->getTargetExcludesByGroup($groupName, Group::EXTENDS_KEY);
+
+        $rule = PHPat::rule()->classes(...self::getPHPSelectors($includes));
+
+        $excludes = self::getPHPSelectors($excludes);
+        if ($excludes !== []) {
+            $rule = $rule->excluding(...$excludes);
+        }
+
+        $rule = $rule->shouldExtend()->classes(...self::getPHPSelectors($extensions));
 
-    public function __construct(
-        public Selectable $selector,
-        public Selectable $parent,
-    ) {
+        $extensionExcludes = self::getPHPSelectors($extensionExcludes);
+        if ($extensionExcludes !== []) {
+            $rule = $rule->excluding(...$extensionExcludes);
+        }
+
+        return $rule->because("$groupName should extend class.");
     }
 
     /**
-     * @param array<string, mixed> $data
+     * @param iterable<Selectable> $selectors
      *
-     * @throws JsonException
+     * @return iterable<Selectable>
      */
-    public static function fromArray(Selectable $selector, array $data): self
+    private static function checkIfNotInterfaceSelectors(iterable $selectors): iterable
     {
-        $parent = Selectors::findSelector($data);
-
-        if ($parent instanceof InterfaceClassSelector) {
-            throw new InvalidArgumentException(
-                'The parent class must not be an interface.'
-            );
+        foreach ($selectors as $selector) {
+            if ($selector instanceof InterfaceClassSelector) {
+                throw new InvalidArgumentException(
+                    "$selector->interface cannot be used in the MustExtend rule, as it is an interface."
+                );
+            }
+            yield $selector;
```

Comment:
Could also just return `$selectors` after the `foreach` instead of re-`yield`ing